ir/type.cpp

```diff
@@ -789,6 +786,10 @@ AggregateType::extract(const StateValue &val, unsigned index, bool fromInt)
 }
 
 unsigned AggregateType::bits() const {
+  if (elements == 0)
+    // It is set as 1 because zero-width bitvector is invalid.
+    return 1;
```

Comment:
uhm, doesn't this have an impact on stores? Storing an empty struct is an no-op, right?  But not if the size is 1 bit, I guess.

Reply:
I think it is okay - it's because an aggregate is elementwisely stored.
I added a relevant test for this.